lib/spacesuit/auth_middleware.ex

```diff
@@ -4,21 +4,23 @@ defmodule Spacesuit.AuthMiddleware do
 
   @timed(key: "timed.authMiddleware-handle", units: :millisecond)
   def execute(req, env) do
-    session_service = Application.get_env(:spacesuit, :session_service)
-
     case req[:headers]["authorization"] do
       nil ->
         Logger.debug "No auth header"
         {:ok, req, env}
 
       "Bearer " <> token ->
-        case session_service[:enabled] do
-          true ->
-            session_service[:impl].handle_bearer_token(req, env, token, session_service[:url])
-          false -> 
+        with {true, :enabled?}      <- enabled?(),
```

Comment:
If we configure this in the routes config, we can move all this code into a function and get it out of this pattern match. Multi-level pattern matching is hard to work with/understand.